Add initiative for product (process and feature ownership) gaps (#1022)
sourcegraph/handbook: content/product-engineering/product/index.md

- Owner: Christina
- [Working Document](https://docs.google.com/document/d/1L-TnZjcYNjwTo2fqUF3DOfUvznbfwCAYvDYLBwwqJW8/edit?usp=sharing)

### Document known product gaps and process

> THere are various areas without a clear owner: we need a list of those, what people should do in the meantime for each of them, and how to flag if somoene thinks they have found a product gap.

Author: ```suggestion
THere are various problems, feature areas, or initiatives related to Product (and sometimes Engineering or other departments) without a clear owner: we need a list of those, what people should do in the meantime for each of them if we are not actively working on solving them, and how to report or add one if someone thinks they have found a new product gap.
```

> THere are various problems, feature areas, or initiatives related to Product (and sometimes Engineering or other departments) without a clear owner: we need a list of those, what people should do in the meantime for each of them if we are not actively working on solving them, and how to report or add one if someone thinks they have found a new product gap.

Author: ```suggestion
There are various problems, feature areas, or initiatives related to Product (and sometimes Engineering or other departments) without a clear owner: we need a list of those, what people should do in the meantime for each of them if we are not actively working on solving them, and how to report or add one if someone thinks they have found a new product gap.
```

> There are various problems, feature areas, or initiatives related to Product (and sometimes Engineering or other departments) without a clear owner: we need a list of those, what people should do in the meantime for each of them if we are not actively working on solving them, and how to report or add one if someone thinks they have found a new product gap.

Author: ```suggestion
There are various problems, feature areas, or initiatives related to Product (which sometimes touch on Engineering or other departments) without a clear owner: we need a list of those, what people should do in the meantime for each of them (if we are not actively working on solving them), a link to a ticket or other resource where the work is happening to fix it, and how to report or add one if someone thinks they have found a new ownership gap.
```
